### PR TITLE
New LaunchOptions properties DefaultViewport and IgnoredDefaultArgs

### LINKED ISSUE #565: Add option to specify the default viewport
https://github.com/kblok/puppeteer/pull/10/commits/25632133e2683c613c97d9088968c48e3c07a086

### LINKED ISSUE #566: Allow options to be passed into puppeteer.defaultArgs
https://github.com/kblok/puppeteer/pull/10/commits/6825088644b26ac1f2dec0230047c0e7314fc4df

### LINKED ISSUE #567: ignoreDefaultArgs to accept array of options
https://github.com/kblok/puppeteer/pull/10/commits/40466cb3a45312986ed126d932cec896c9f13dbe

### PR DESCRIPTION
**Pull request by kblok**
closes #567
closes #566 
closes #565

I found many differences related to DefaultView port. I don't know if we missed them or those are upcoming changes, but I took the chance and change them.

### CODE REVIEW

**Comment by kblok** [comment]
Ready to review @Meir017.
I had to implement 3 issues in 1 because all three were related to argument changes.

File: lib/PuppeteerSharp/ChromiumProcess.cs [hunk 1]
```diff
@@ -205,69 +202,71 @@ public async Task<bool> WaitForExitAsync(TimeSpan? timeout)
...
+            var chromiumArgs = new List<string>();
+
+            if (!options.IgnoreDefaultArgs)
             {
-                options.Headless = false;
+                chromiumArgs.AddRange(GetDefaultArgs(options));
+            }
+            else if (options.IgnoredDefaultArgs?.Length > 0)
+            {
+                chromiumArgs.AddRange(GetDefaultArgs(options).Where(a => !options.IgnoredDefaultArgs.Contains(a)));
```
Comment (Meir017) [inline lib/PuppeteerSharp/ChromiumProcess.cs]: just use `GetDefaultArgs(options).Except(options.IgnoredDefaultArgs)`

Comment (Meir017) [inline lib/PuppeteerSharp/ChromiumProcess.cs]: just use `GetDefaultArgs(options).Except(options.IgnoredDefaultArgs)`

File: lib/PuppeteerSharp/ConnectOptions.cs [hunk 2]
```diff
@@ -41,5 +52,11 @@ public class ConnectOptions : IBrowserOptions, IConnectionOptions
         /// Optional factory for <see cref="WebSocket"/> implementations.
         /// </summary>
         public Func<Uri, IConnectionOptions, CancellationToken, Task<WebSocket>> WebSocketFactory { get; set; }
+
+        /// <summary>
+        /// Gets or sets the default Viewport.
+        /// </summary>
+        /// <value>The default Viewport.</value>
+        public ViewPortOptions DefaultViewport { get; set; }
```
Comment (Meir017) [inline lib/PuppeteerSharp/ConnectOptions.cs]: the default value can be set here
```cs
public ViewPortOptions DefaultViewport { get; set; } = new ViewPortOptions
{
    Width = 800,
    Height = 600
};
```

File: lib/PuppeteerSharp/LaunchOptions.cs [hunk 3]
```diff
@@ -11,19 +11,24 @@ namespace PuppeteerSharp
...
+        private string[] _ignoredDefaultArgs;
+
         /// <summary>
-        /// Whether to ignore HTTPS errors during navigation. Defaults to false.
+        /// Initializes a new instance of the <see cref="T:PuppeteerSharp.LaunchOptions"/> class.
         /// </summary>
-        public bool IgnoreHTTPSErrors { get; set; }
+        public LaunchOptions()
+        {
+            DefaultViewport = new ViewPortOptions
```
Comment (Meir017) [inline lib/PuppeteerSharp/LaunchOptions.cs]: this code appears also in the `ConnectOptions` ctor, maybe add a `Default` static readonly property on the `ViewPortOptions` class?

**Comment by kblok** [comment]
Done @Meir017

File: lib/PuppeteerSharp/ViewPortOptions.cs [hunk 4]
```diff
@@ -45,5 +45,14 @@ public ViewPortOptions()
         /// </summary>
         /// <value>Specifies if viewport supports touch events. Defaults to <c>false</c>.</value>
         public bool HasTouch { get; set; }
+
+        /// <summary>
+        /// Default Viewport
+        /// </summary>
+        public static ViewPortOptions Default => new ViewPortOptions
```
Comment (Meir017) [inline lib/PuppeteerSharp/ViewPortOptions.cs]: make readonly

Comment (kblok) [reply lib/PuppeteerSharp/ViewPortOptions.cs]: But it's a getter. It is read-only. Am I missing something?

Comment (Meir017) [reply lib/PuppeteerSharp/ViewPortOptions.cs]: Oops

**Pull request merged**